### ISSUE TITLE
Mijia Smart Clothes Drying Rack

### ISSUE THREAD

**Issue by makleso6** [jghaanstra/com.xiaomi-miio#276]
> As there is no official API and Xiaomi does not support third-party integration, being able to add a device to the Homey app depends on an existing integration, online documentation or donating the device to the developer. If neither of these can be fulfilled then integration is not possible.

### Supply the device details
* **Name:** Mijia Smart Clothes Drying Rack
* **Model ID:** hyd.airer.znlyj4
* **MIoT spec:** https://home.miot-spec.com/spec/hyd.airer.znlyj4

### Current support
no support

### Additional information
![IMG_6456](https://github.com/user-attachments/assets/4bb448c8-62dd-4505-90bd-abea270200c4)
it can be window covering with lamp

---- STOP EDITING HERE----
### Donation
I enjoy creating apps for Homey and try to support them as best as I can. If you enjoy using my apps a [donation](https://www.paypal.me/jghaanstra) in return for the time I put into this is much appreciated.

**Comment by makleso6** [comment]
also I can try to add it by pr, but first you need to figure out the Homey development tools

**Comment by jghaanstra** [comment]
Added with release 3.2.29 which is currently pending approval. Please test.

https://homey.app/a/com.xiaomi-miio/test/

**Comment by makleso6** [comment]
I have errors
![IMG_6470](https://github.com/user-attachments/assets/dc48031a-fff5-42c1-a890-e0ab566c6633)

**Comment by makleso6** [comment]
also bug in temp for zhimi.airfresh.va4
(2.18 instead 21.8)
![IMG_6469](https://github.com/user-attachments/assets/e023d31d-05e1-4001-8865-9a0a7f93e20d)

**Comment by makleso6** [comment]
scale should be only for 'zhimi.airfresh.va2'
https://github.com/jghaanstra/com.xiaomi-miio/pull/275/files#diff-d041edde56fd8ca037808dc79894e5330ad49a3d119b2223d0d4658da97a48c5R135

**Comment by makleso6** [comment]
crate report 17395ddf-9d99-40b5-9724-2ef44ce761bc

**Comment by jghaanstra** [comment]
Try release 3.2.30 which is pending approval now.

https://homey.app/a/com.xiaomi-miio/test/

**Comment by makleso6** [comment]
can not control up/down
![IMG_6475](https://github.com/user-attachments/assets/694b0fc7-1216-490f-8e2e-fcb884ae9553)

**Comment by makleso6** [comment]
![IMG_6475](https://github.com/user-attachments/assets/c317ad67-ebb4-48ab-92f9-8a06e2d669e3)

**Comment by makleso6** [comment]
how can I run app on my Homey (to test rack)
I have error
```
✓ Uninstalling `com.xiaomi-miio`...
✓ Removing container...
node:internal/process/promises:392
      new UnhandledPromiseRejection(reason);
      ^

UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "Cannot find module './common'
Require stack:
- /app/node_modules/debug/src/node.js
- /app/node_modules/debug/src/index.js
- /app/node_modules/tinkerhub-discovery/lib/abstract.js
- /app/node_modules/tinkerhub-discovery/index.js
- /app/node_modules/miio/lib/discovery.js
- /app/node_modules/miio/lib/index.js
- /app/app.js
- /node_modules/@athombv/homey-apps-sdk-v3/lib/Util.js
- /node_modules/@athombv/homey-apps-sdk-v3/lib/Device.js
- /node_modules/@athombv/homey-apps-sdk-v3/homey.js
- /node_modules/@athombv/homey-apps-sdk-v3/index.js
- /homey-app-runner/lib/App.js
- /homey-app-runner/index.js".
    at throwUnhandledRejectionsMode (node:internal/process/promises:392:7)
    at processPromiseRejections (node:internal/process/promises:475:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:106:32) {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

**Comment by jghaanstra** [comment]
Try `npm install` in the folder of your local fork.

**Comment by makleso6** [comment]
thanks! a run app 
see there is no set_properties for controls
could I made a pr?

**Comment by jghaanstra** [comment]
A fix has been added with release 3.2.31 which is pending approval. Please test it.

https://homey.app/a/com.xiaomi-miio/test/

**Comment by makleso6** [comment]
Yes it works. I can up/down/pause and on/off light
